### PR TITLE
Add Data-to-Text Generation, Language Modeling, and Mathematical Reasoning Scenarios

### PR DESCRIPTION
**Pull request by Jianqiao-Zhao**
Including related subtasks and metrics.

### CODE REVIEW

File: src/helm/benchmark/metrics/basic_metrics.py [hunk 1]
```diff
@@ -278,6 +278,22 @@ def tokenize(self, text: str) -> List[str]:
...
+def chinese_math_exact_match_indicator(gold: str, pred: str) -> float:
+    """
+    Exact match that only cares the last math expression.
+    Common math expressions are numbers and fractions.
+    """
+    pattern = r"[-+*/%\.\(\)\d]+"
+    matches = re.findall(pattern, pred)
+    if matches:
+        pred = matches[-1].lstrip(")")
+    for term in ["\n", "。", "，"]:
```
Comment (lyy1994) [inline src/helm/benchmark/metrics/basic_metrics.py]: You can remove the truncation here, as HELM will perform it in adaptation (via `stop_sequences`).

File: src/helm/benchmark/metrics/basic_metrics.py [hunk 2]
```diff
@@ -278,6 +278,22 @@ def tokenize(self, text: str) -> List[str]:
     return partial(rouge_score, scorer=scorer, rouge_type=rouge_type)
 
 
+def chinese_math_exact_match_indicator(gold: str, pred: str) -> float:
```
Comment (lyy1994) [inline src/helm/benchmark/metrics/basic_metrics.py]: I recommend a different name: we are not using indicator here.

File: src/helm/benchmark/scenarios/cleva_scenario.py [hunk 3]
```diff
@@ -1105,3 +1105,132 @@ class CLEVAReasoningPrimitiveScenario(CLEVAScenario):
...
+            references=references,
+            split=split,
+        )
+        return instance
+
+
+class CLEVALanguageModelingScenario(CLEVAScenario):
+    """
+    The language modeling task of CLEVA benchmark.
+    The data comes from news (https://github.com/brightmart/nlp_chinese_corpus).
```
Comment (lyy1994) [inline src/helm/benchmark/scenarios/cleva_scenario.py]: Should we put this information into the comment? I think we may switch to another text corpus in the future and our code should not tie to a specific dataset. But we can state that the example comes from the mentioned dataset.

File: src/helm/benchmark/metrics/basic_metrics.py (hunk 1 above)
Comment (Jianqiao-Zhao) [reply src/helm/benchmark/metrics/basic_metrics.py]: Adjusted.

File: src/helm/benchmark/metrics/basic_metrics.py (hunk 2 above)
Comment (Jianqiao-Zhao) [reply src/helm/benchmark/metrics/basic_metrics.py]: I changed the name to `chinese_math_result_match`

File: src/helm/benchmark/scenarios/cleva_scenario.py (hunk 3 above)
Comment (Jianqiao-Zhao) [reply src/helm/benchmark/scenarios/cleva_scenario.py]: I rewrote the comment. The new version is more general.

**Review by lyy1994 — APPROVED** [review]

**Pull request merged**